e2e/testcases/admission_test.go

```diff
@@ -141,55 +148,84 @@ metadata:
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		return nt.Validate(
+			core.RootReconcilerName(configsync.RootSyncName),
+			configsync.ControllerNamespace,
+			&appsv1.Deployment{},
+			testpredicates.DeploymentMissingEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled),
+		)
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
 	}
-
-	// Remove the annotation for disabling webhook
-	if _, err := nt.Shell.Kubectl("annotate", "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io", "configsync.gke.io/webhook-configuration-update-"); err != nil {
-		nt.T.Fatalf("failed to remove the annotation in the admission webhook %v", err)
+	nt.T.Logf("Check if the namespace resourceversion has changed after deleting Admission Webhook")
+	err := nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+		[]testpredicates.Predicate{
+			testpredicates.ResourceVersionNotEquals(nt.Scheme, version),
+		})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Logf("Check if the test namespace has dropped the declared-field annotation")
+	if err := nt.Validate("hello", "", &corev1.Namespace{}, testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey)); err != nil {
+		nt.T.Fatal(err)
 	}
 
-	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/test/ns.yaml", fake.NamespaceObject("test")))
-	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add another Namespace"))
-	if err := nt.WatchForAllSyncs(); err != nil {
+	helloNS = &corev1.Namespace{}
+	if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
 		nt.T.Fatal(err)
 	}
+	version = helloNS.ResourceVersion
 
+	if err := nomostest.InstallWebhook(nt); err != nil {
+		nt.T.Fatal(err)
+	}
 	nomostest.WaitForWebhookReadiness(nt)
 
-	// Verify that the webhook is now enabled
-	if _, err := nt.Shell.Kubectl("delete", "ns", "test"); err == nil {
-		nt.T.Fatal("got `kubectl delete ns hello` success, want return err")
+	nt.T.Logf("Check if the namespace resourceversion has changed after deleting Admission Webhook")
+	tg = taskgroup.New()
+	tg.Go(func() error {
+		return nt.Validate(
+			core.RootReconcilerName(configsync.RootSyncName),
+			configsync.ControllerNamespace,
+			&appsv1.Deployment{},
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled, "true"),
+		)
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Logf("Check if the namespace resourceversion has changed after installing Admission Webhook")
+	err = nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+		[]testpredicates.Predicate{
+			testpredicates.ResourceVersionNotEquals(nt.Scheme, version),
+		})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Logf("Check if the test namespace has restored the declared-field annotation")
+	if err := nt.Validate("hello", "", &corev1.Namespace{}, testpredicates.HasAnnotationKey(metadata.DeclaredFieldsKey)); err != nil {
```

Comment:
Similarly, this validation can be merged with `nt.Watcher.WatchObject`.

Reply:
Done